include/hipSYCL/compiler/Frontend.hpp

```diff
@@ -171,7 +171,7 @@ inline std::string buildKernelName(clang::RecordDecl* D, clang::MangleContext *M
 }
 
 // Partially taken from CGCUDANV.cpp
-#if LLVM_VERSION_MAJOR >= 13
+#if LLVM_VERSION_MAJOR >= 13 || !defined(HIPSYCL_NO_DEVICE_MANGLER)
```

Comment:
I think this needs to be `&&` not `||`. Otherwise you would be disabling explicit multipass whenever that option is not defined. What you want is: Enable explicit multipass if we have at least clang 13 *and* we have not been explicitly told to disable it. Or am I missing something?